src/docs/patterns/forms.mdx

```diff
@@ -82,3 +95,78 @@ In some specific situations, you may need to defer verification or perform it fo
         '2: Use a circular progress indicator when waiting for responses from the server.',
     ]}
 />
+
+## Submit the Form
+
+Usually, a form will have a submit/next/finished button at the end. This button is typically placed at the bottom right corner of the form. If you are designing for a right-to-left language, this button should be at the bottom left.
+
+<ImageGrid
+    images={[SumbitButtonLocationImage]}
+    gridImageProps={{ sm: 12 }}
+    caption={[
+        '1: On desktops, the button is placed in the bottom corner of the form.',
+        '2: On mobile devices, the button is occluded by the native keyboard.',
+    ]}
+/>
+<ImageGrid
+    images={[NextButtonUsageImage]}
+    gridImageProps={{ sm: 12 }}
+    caption={[
+        'On mobile, if there are more than one input fields on the screen, the keyboard button should say "next".',
+        '1: The user entered a valid input. Clicking "next" will autofocus the user to the next text field.',
+        '2: The user entered a invalid input. Clicking "next" will display the error message but will not autofocus to the next text field.'
+    ]}
+/>
+
+### Disable the Submit Button
+
+When there are not a lot of text fields, the submit button can be disabled to indicate that there are some errors in the user input.
+
+<ImageGrid
+    images={[SubmitButtonOnChangeImage]}
+    gridImageProps={{ sm: 12 }}
+    caption={
+        'If you have only one input field, and this field is verified on change, disable the submit button until the user entered the desired input.'
+    }
+/>
+<ImageGrid
+    images={[SubmitButtonAutoSubmitImage]}
+    gridImageProps={{ sm: 12 }}
+    caption={
+        'For the verification code screen, you can auto-submit the code once the user input reaches the desired length (2). After the verification code is recognized (3), the screen pauses for a short moment before auto-transit to the next page. The verify button is preserved for consistency.'
+    }
+/>
+
+If at least one text field uses the 'verify on blur' strategy, do not disable the submit button. Otherwise, the mobile user will have to disable their keyboard first to have their input field verified, before they can click the submit button. The only exception is if there are required input fields.
+
+<ImageGrid
+    images={[SubmitButtonOnBlurImage]}
+    gridImageProps={{ sm: 12 }}
+    caption={'The submit button is disabled until the user has entered something for all the required fields.'}
+/>
+
+When you have a long form (more than three text fields) with a lot of text fields and the user forgets a required field, do not auto disable the submit button -- it will take them some effort to check what they are missing. Instead, allow the user to click the submit button, and highlight the errors for them.
```

Comment:
Seems like a lot of conflicting advice. Above we say "use onBlur for most things", then we say "if you're using on blur, don't disable the submit button", and then we say "actually, disable the submit button if you have required fields", and now we say "but if you have a long form, don't disable it even if you have required fields".

Maybe we could have some small example sections instead of trying to be too generic. E.g., "This form has two fields, one of which is required. THIS IS THE BEHAVIOR YOU SHOULD IMPLEMENT IN THIS SITUATION." "This is a multi-page form (first page) and all fields are optional. THIS IS THE BEHAVIOR YOU SHOULD IMPLEMENT IN THIS SITUATION."

Reply:
yeah, it is complicated. I'd need a huge flow chart for that 😛